specs/core/0_beacon-chain.md

```diff
@@ -1048,18 +1077,27 @@ Let `genesis_block = BeaconBlock(state_root=hash_tree_root(genesis_state))`.
 The post-state corresponding to a pre-state `state` and a block `block` is defined as `state_transition(state, block)`. State transitions that trigger an unhandled exception (e.g. a failed `assert` or an out-of-range list access) are considered invalid.
 
 ```python
-def state_transition(state: BeaconState, block: BeaconBlock, validate_state_root: bool=False) -> BeaconState:
+def state_transition(state: BeaconState, signed_block: SignedBeaconBlock, checked: bool=True) -> BeaconState:
```

Comment:
The signature check is dependent on the expected state root (part of the block, thus signed over). This is why it doesn't make much sense to have a way to disable the state-root check, like previously, without disabling the signature.
If we want to be more explicit, I think it makes more sense to call it `validate_result` or something like that. Naming things is difficult :sweat_smile:

Reply:
Aah sorry I misread it!
IMO `checked` sounds like it has been checked *before* this function and `validate_result` is more accurate. :)

FYI there is no `checked=False` in this branch (including tests), but I think it's fine to keep it in case we end up needing it in the future...?